core/src/main/java/com/linecorp/armeria/common/util/PentaFunction.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.common.util;
+
+/**
+ * A custom Function interface which takes five argument.
+ */
+@FunctionalInterface
+public interface PentaFunction<A, B, C, D, E, F> {
```

Comment:
Instead of introducing a generic type, how about introducing a package-private interface dedicated for accepting the constructor, like:

```java
@FunctionalInterface
interface ThreadFactoryProvider {
    ThreadFactory get(String, boolean, int, Function<? super Runnable, ? extends Runnable>);
}
```

..?

Reply:
Added `TheadFactoryProvider`
Removed `PentaFunction`
Thank you.